Bug: Hyperliquid size precision causes float_to_wire rounding error

## Bug

When placing live trades on Hyperliquid, position sizes with too many decimal places cause a `ValueError: float_to_wire causes rounding` from the SDK.

## Reproduction

1. Run a live strategy with `capital_pct` or a capital that results in a non-round position size
2. Signal triggers BUY/SELL
3. Size calculates to e.g. `0.4671228074779094`
4. SDK rejects it: `ValueError: ("float_to_wire causes rounding", 0.4671228074779094)`

## Expected Behavior

The adapter should round the size to the correct tick precision before sending to the SDK. ETH on Hyperliquid uses 0.001 tick size.

## Fix

In `platforms/hyperliquid/adapter.py`, round the size to the appropriate precision (e.g. 3 decimal places for ETH) before calling `market_open` / `market_close`.

```python
# Before calling exchange
size = round(size, 3)  # or use a per-symbol tick size lookup
```

## Log

```
[INFO] Placing live buy ETH size=0.467123
[ERROR] Live execute returned error: ('float_to_wire causes rounding', 0.4671228074779094)
```

--- a/platforms/hyperliquid/adapter.py
+++ b/platforms/hyperliquid/adapter.py
@@ -201,6 +201,11 @@ class HyperliquidExchangeAdapter:
             raise RuntimeError(
                 "market_open requires live mode (set HYPERLIQUID_SECRET_KEY)"
             )
+        # Round to asset's tick precision to avoid float_to_wire rounding error
+        sz_decimals = self._info.asset_to_sz_decimals.get(symbol, 3)
+        size = round(size, sz_decimals)
+        if size <= 0:
+            raise ValueError(f"Size rounded to zero for {symbol} (sz_decimals={sz_decimals})")
         return self._exchange.market_open(symbol, is_buy, size, None, 0.01)
 
     def market_close(self, symbol: str) -> dict:

